### PR TITLE
sumologicexporter: do not overwrite attributes using metadata translation

### PR DESCRIPTION
**Pull request by sumo-drosiek**
User should be able to write it owns attributes (especially _source ones) and we shouldn't overwrite it

### CODE REVIEW

File: pkg/exporter/sumologicexporter/translate_metadata.go [hunk 1]
```diff
@@ -42,12 +42,16 @@ var metadataTranslations = map[string]string{
...
+	"file.path.resolved":      "_sourceName",
 }
 
 // translateMetadata renames metadata keys according to metadataTranslations.
 func translateMetadata(attributes pdata.AttributeMap) {
 	attributes.Range(func(otKey string, value pdata.AttributeValue) bool {
 		if sumoKey, ok := metadataTranslations[otKey]; ok {
+			if _, ok := attributes.Get(sumoKey); ok {
+				return true
+			}
```
Comment (pmalek-sumo) [inline pkg/exporter/sumologicexporter/translate_metadata.go:54]: Can we add some comment here? It's not obvious to me what this does.

For all attributes, for all all metadata translated attributes check if particular sumo metadata key is in the metric attributes and if it is then skip it (don't remove it?)?

Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/translate_metadata.go:55]: I added comment

Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/translate_metadata.go:55]: `// do not rename attribute if target name already exists`

Comment (pmalek-sumo) [reply pkg/exporter/sumologicexporter/translate_metadata.go:55]: Sorry but I don't understand: `do not rename attribute if target name already exists` -> this checks if `sumoKey` is in `attributes`. Without this check we simply overwrite the value under this key (and remove the corresponding OT key).

Maybe this should be using a different wording? If it exists do not change the value of already existing Sumo key and do not remove a corresponding OT key ?

Comment (sumo-drosiek) [reply pkg/exporter/sumologicexporter/translate_metadata.go:55]: we can use different wording, I understand it that way.

```
			attributes.Insert(sumoKey, value)
			attributes.Delete(otKey)
```
renames (move) attribute from `otKey` to `sumoKey`

if attribute with name `sumoKey` (target) already exists do not overwrite it

**Review by pmalek-sumo — APPROVED** [review]

File: pkg/exporter/sumologicexporter/translate_metadata_test.go [hunk 2]
```diff
@@ -62,19 +62,19 @@ func TestTranslateMetadataLeavesOtherAttributesUnchanged(t *testing.T) {
 	assertAttribute(t, metadata, "three", "three1")
 }
 
-func TestTranslateMetadataOverwritesExistingAttribute(t *testing.T) {
+func TestTranslateMetadataDoNotOverwritesExistingAttribute(t *testing.T) {
```
Comment (astencel-sumo) [inline pkg/exporter/sumologicexporter/translate_metadata_test.go:65]:
```suggestion
func TestTranslateMetadataDoesNotOverwriteExistingAttribute(t *testing.T) {
```

File: pkg/exporter/sumologicexporter/translate_metadata_test.go [hunk 3]
```diff
@@ -62,19 +62,19 @@ func TestTranslateMetadataLeavesOtherAttributesUnchanged(t *testing.T) {
...
 	translateMetadata(metadata)
 
-	assert.Equal(t, 1, metadata.Len())
-	assertAttribute(t, metadata, "host", "hostname1")
+	assert.Equal(t, 2, metadata.Len())
+	assertAttribute(t, metadata, "host", "host1")
 }
 
-func TestTranslateMetadataOverwritesMultipleExistingAttributes(t *testing.T) {
+func TestTranslateMetadataDoNotOverwritesMultipleExistingAttributes(t *testing.T) {
```
Comment (astencel-sumo) [inline pkg/exporter/sumologicexporter/translate_metadata_test.go:77]:
```suggestion
func TestTranslateMetadataDoesNotOverwriteMultipleExistingAttributes(t *testing.T) {
```

**Pull request merged**